src/paymentservice/charge.js

```diff
@@ -40,29 +57,34 @@ module.exports.charge = async request => {
   });
 
   if (!valid) {
+    span.setAttributes({ 'app.variant': 'A', 'app.loyalty.level': random(LOYALTY_LEVEL) });
     throw new Error('Credit card info is invalid.');
   }
 
   if (!['visa', 'mastercard'].includes(cardType)) {
+    span.setAttributes({ 'app.variant': 'A', 'app.loyalty.level': random(LOYALTY_LEVEL) });
     throw new Error(`Sorry, we cannot process ${cardType} credit cards. Only VISA or MasterCard is accepted.`);
   }
 
   if ((currentYear * 12 + currentMonth) > (year * 12 + month)) {
+    span.setAttributes({ 'app.variant': 'A', 'app.loyalty.level': random(LOYALTY_LEVEL) });
     throw new Error(`The credit card (ending ${lastFourDigits}) expired on ${month}/${year}.`);
   }
 
-  // check baggage for synthetic_request=true, and add charged attribute accordingly
+  // Check baggage for synthetic_request=true, and add charged attribute accordingly
   const baggage = propagation.getBaggage(context.active());
-  if (baggage && baggage.getEntry("synthetic_request") && baggage.getEntry("synthetic_request").value === "true") {
+  if (baggage && baggage.getEntry('synthetic_request') && baggage.getEntry('synthetic_request').value === 'true') {
     span.setAttribute('app.payment.charged', false);
   } else {
     span.setAttribute('app.payment.charged', true);
   }
 
-  span.end();
+  span.setAttributes({ 'app.variant': 'A', 'app.loyalty.level': random(LOYALTY_LEVEL) });
```

Comment:
This `app.loyalty.level` attribute is added similarly, but the code is repeated multiple times across the various scenarios. Can we break this out to add just this attribute by itself at the top of the charge function and let the logic control the other span attribute only?

Reply:
I have removed the repetition of the `setAttributes` and now have the scenario that when the feature flag is set only `gold` level will error. Have introduced another level `platinum` so along with `silver` and `bronze` will show success.

<img width="385" alt="image" src="https://github.com/user-attachments/assets/5d289bb4-7f98-454e-b100-befd60dc8a22" />